### ISSUE TITLE
Добавление в home-assistant?

### ISSUE THREAD

**Issue by foxel** [dmitry-k/yandex_smart_home#16]
Привет.

Как насчет перевода этого компонента из статуса кастомного мода в полноценный официальный компонент?

Хотел сам писать этот компонент, а тут твой нашел. Может помочь с вкладом в официальный HA?

**Comment by dmitry-k** [comment]
Да, наверное попробую подготовить PR в ближайшее время. Я мельком смотрел требования, всё вроде-бы более-менее понятно. Как и то что они не любят большие PR, могу понять)

Но слишком много времени тратить на это тоже не хочется, посмотрю как пойдёт.

**Comment by donatelloth** [comment]
тоже очень жду :)

**Comment by Siorinex** [comment]
Да можно-ж даже не в основном их списке? Там куча полезняшек, типа EspHome, резервирования на Гугл и т.д. идут как подключаемые через Гит...

**Comment by foxel** [comment]
Коммит в основной репо сулит интеграцию с Hass Cloud. Тогда заводить Алису будет совсем просто

**Comment by Taygrim** [comment]
Можно добавить не в основной репозиторий, а в HACS: https://hacs.netlify.com/docs/publish/integration. Насколько я понял, для этого достаточно поместить код в custom_components/yandex_smart_home

**Comment by pruwait** [comment]
Тоже хочу установить этот компонент в HASSIO (докер). 
Поставил esphome, добавив git в аддонах. А что добавить для Алисы?
Можно более подробную инструкцию?

**Comment by foxel** [comment]
Надо просто склонировать репо в папку custom_components/yandex_smart_home.
Для этого достаточно имеет установленный аддон ssh

**Comment by pruwait** [comment]
Оказалось все просто.
sudo mkdir /usr/share/hassio/homeassistant/custom_components
cd /usr/share/hassio/homeassistant/custom_components
sudo git clone https://github.com/dmitry-k/yandex_smart_home.git
Далее прописываем в configuration.yaml этот самый компонент, просто добавляем в конце файла "yandex_smart_home:" (без кавычек, конечно);

**Comment by dmitry-k** [comment]
В для официального так и не собрался и вряд ли соберусь написать тесты.  Добавил в HACS.

**Comment by jlisck** [comment]
Добрый день!
У меня возникла проблема, при добавлении телевизоров и проигрователей на kodi не доступно управление питанием, устройства не выключаются в яндекс утройствах кнопка питания не активна, при попытке выключить громкость на kodi говорит "в установленном режиме команда не потдерживается", ранее ставил ваш плагин все хорошо работало без доаолнительных настроек. Подскажите пожалуйста как поправить.

**Comment by goldenman1984** [comment]
Не смог добавить cover. Там какие то особенности для добавления?

**Comment by pruwait** [comment]
А есть возможность из хассио управлять навыками, которые уже есть у Алисы? Хотелось управлять устройствами owencloud

**Comment by mrloggyst** [comment]
Доброй ночи. Что-то не получается у меня завести. Пишет что не может найти интеграцию причем ручками созданная тестовая в соседнем каталоге видится (. Home Assistant на Raspbian pi4 в docker-е.

**Comment by kontsantin** [comment]
Здравствуйте. У мне серьёзная проблема, не могу создать безопасное соединение через ssh для интеграции яндекс станции в HA, через putty не получается выдаёт ошибку при вводе команды "sudo apt-get install certbot" не существует такой команды.

**Comment by PaulAnnekov** [comment]
@dmitry-k я так понимаю, что вы сейчас не работаете над PR в HA core и вы не против, если кто-то вместо вас это сделает с упоминанием вас в качестве codeowner в manifest.json компонента?

**Comment by dmitry-k** [comment]
Да и да.
Для начала надо написать тесты. Насколько я помню, они там обязательны.
И после включения будет достаточно сложно вносить изменения. Мне и тут-то уже лень что-то делать, а там и подавно будет. Здесь, в отличие от большинства компонентов, всё сильно зависит от HA и нельзя добавить в HA тонкую прослойку, а основной код хранить где-то в своём репозитории.
Но зато будет меньше шанс того что всё сломается при очередном обновлении HA.

**Comment by dext0r** [comment]
Для будущей версии 0.2.0 полностью перетряхнул код и написал тесты со 100% покрытием. 
Однако, мне бы хотелось, чтобы компонент оставался независимым от релизного цикла HA. Тем более что через HACS всё прекрасно ставится и удобно обновляется.

**Comment by foxel** [comment]
@dext0r @dmitry-k два пункта к обсуждению в данном контексте:
1 релизный цикл HA может вносить изменения во внутренние структуры, которые используются в данном расширении
2 имея этот код в HA можно дотащить интеграцию до HA cloud/Nabu Casa, чтобы не приходилось создавать аккаунт в Yandex и сделать компонент доступным рядовым пользователям HA

**Comment by dext0r** [comment]
1. Тесты должны бежать по нескольким версиям HA, поэтому здесь никакой проблемы нет.
2. Сначала нужно понимать, какой процент людей из стран, где работает полноценно работает УДЯ, готов платить 5$ в месяц за облачную подписку на HA. А еще недостаточно просто протащить компонент в ядро, нужно убедить команду HA Cloud добавить поддержку компонента на серверную сторону (код которой не в OpenSource).

**Comment by PaulAnnekov** [comment]
Я из Украины, у нас Яндекс заблокирован на вход и выход. Мне пришлось настраивать прокси сервер в другой стране, который перенаправляет запросы от и к Яндексу. А потом ещё мучаться, настраивать навык в Яндексе, который бы с этим работал. Если бы nabucasa поддерживала УДЯ, я бы заплатил. 

Но это мой специфичный кейс)).

**Comment by foxel** [comment]
@dext0r 
 1. бывает что версия ломает все внутри, соответственно для "старого" и "нового" HA должны быть разные версии кода. Разрабатываю модуль внутри HA и парочку для личного пользования - напарывался в последний год. Возможно HACS как-то в этом помогает, например через маппинг совместимых версий.
 2. Думаю достаточно, учитывая, что Nabu Casa дает не только это но и DNS. По поводу убеждения - основные компоненты OSS, у них прокси там в закрытом коде ну и регистрация в облаке yandex от них потребуется. Смотрим [тут](https://github.com/home-assistant/core/blob/27e29b714cacadf308f983e8ae2dbc8f5e010a83/homeassistant/components/cloud/client.py#L187) и [тут](https://github.com/NabuCasa/hass-nabucasa/blob/master/hass_nabucasa/iot.py#L34-L134). Учитывая, что это потенциально принесет им новых подписчиков, думаю тут проблем не будет.

**Comment by interm21** [comment]
Приветствую!
При выполнении шага:
В Home Assistant: Настройки > Интеграции > Добавить интеграцию > Yandex Smart Home
Сразу ошибка:
Отменено
Конфигурация не найдена или содержит ошибки. Проверьте журнал сервера Home Assistant.

**Comment by vismuthxx** [comment]
Вопрос все датчики, лампочки, берет не из файла /config/configuration.yaml
А с страницы Яндекс диалоги
Тогда зачем все что есть в доме в носить в файл для яндекса /config/configuration.yaml?
Объясните пожалуйста
